Modules/pyexpat.c

```diff
@@ -1640,18 +1640,52 @@ add_submodule(PyObject *mod, const char *fullname)
         Py_DECREF(mod_name);
         return NULL;
     }
+    Py_DECREF(mod_name);
 
     /* gives away the reference to the submodule */
     if (PyModule_AddObject(mod, name, submodule) < 0) {
         Py_DECREF(submodule);
-        Py_DECREF(mod_name);
         return NULL;
     }
 
-    Py_DECREF(mod_name);
     return submodule;
 }
 
+static int
+add_error(PyObject *errors_module, PyObject *codes_dict,
+          PyObject *rev_codes_dict, const char *name, int value)
+{
+    const char *error_string = XML_ErrorString(value);
+    if (PyModule_AddStringConstant(errors_module, name, error_string) < 0) {
+        return -1;
+    }
+
+    PyObject *num = PyLong_FromLong(value);
+    if (num == NULL) {
+        return -1;
+    }
+
+    if (PyDict_SetItemString(codes_dict, error_string, num) < 0) {
+        Py_DECREF(num);
+        return -1;
+    }
+
+    PyObject *str = PyUnicode_FromString(error_string);
+    if (str == NULL) {
+        Py_DECREF(num);
+        return -1;
+    }
+
+    int res = PyDict_SetItem(rev_codes_dict, num, str);
+    Py_CLEAR(str);
+    Py_CLEAR(num);
```

Comment:
Sorry, old micro-optimization habits!

```suggestion
    Py_DECREF(str);
    Py_DECREF(num);
```